HAR-3157 näkyykö tarkastus tilaajalle

Jarzka opened the pull request.

Jarzka committed:
Lisää migraatiossa tarkastus-tauluun uusi sarake nayta_urakoitsijalle

Jarzka committed:
Lisää tarkastuksen lomakkeelle checkbox "Näytä urakoitsijalle"

Jarzka committed:
Älä palauta urakoitsijalle näkymättömiä tarkastuksia

Jarzka commented on src/cljs/harja/views/murupolku.cljs:
```diff
@@ -137,10 +138,9 @@
-              urakoitsija? (-> @istunto/kayttaja
-                               :organisaatio
-                               :tyyppi
-                               (= :urakoitsija))]
```
Tämä on ymmärtääkseni toiminut väärin, sillä organisaation tyyppi on aina string.

Jarzka committed:
Urakoitsijan tekemässä tarkastuksessa näytä urakoitsijalle -arvo on aina tosi, muuten voi itse muokata

Jarzka committed:
Korjaa virheitä failaavien testien mukaisesti

Jarzka committed:
Refactoroi if-logiikkaa

Jarzka commented on src/clj/harja/kyselyt/laatupoikkeamat.clj:
```diff
@@ -24,7 +24,7 @@
                                                 (konv/sql-timestamp aika)
                                                 (when tekija (name tekija))
                                                 kohde
-                                                (if selvitys-pyydetty true false)
```
Sama asia, kysytään onko arvo loogisesti totta (false ja nil on false ja kaikki muu true)

Jarzka commented on src/clj/harja/kyselyt/laatupoikkeamat.clj:
```diff
@@ -24,7 +24,7 @@
                                                 (when tekija (name tekija))
                                                 kohde
-                                                (if selvitys-pyydetty true false)
+                                                (boolean selvitys-pyydetty)
```
Sama asia lyhyemmin, kysytään onko arvo loogisesti totta (false ja nil on false ja kaikki muu true)

jarnovayrynen commented on src/clj/harja/palvelin/palvelut/laadunseuranta.clj:
```diff
@@ -214,19 +214,26 @@
+                              (boolean tienumero) tienumero
+                              (boolean tyyppi ) (and tyyppi (name tyyppi))
+                              vain-laadunalitukset?
+                              max-rivimaara))
```
urakoitsija? voisi mennä sql kyselylle asti parametrinä

jarnovayrynen replied:
Kun haetaan yksittäinen tarkastus, pitää myös tarkistaa saako nähdä. muuten voi esim id:llä hakea

Jarzka replied:
Nämä korjattu

Jarzka committed:
Merge branch 'develop' into HAR-3157

# Conflicts:
#	src/clj/harja/kyselyt/tarkastukset.sql
#	tietokanta/src/main/resources/db/migration/V1_429__.sql

Jarzka committed:
Filtteröi vain urakoitsijalle näkyvät tarkastukset suoraan SQL-kyselyssä

Jarzka committed:
Älä palauta tarkastusta id:llä urakoitsijalle jos ei saa nähdä sitä

Jarzka committed:
Päivitä tarkastuksen testit

jarnovayrynen committed:
Merge branch 'develop' into HAR-3157

jarnovayrynen committed:
Merge branch 'develop' into HAR-3157

Jarzka committed:
Korjaa testi

Jarzka committed:
Korjaa testit

tatut committed:
Merge branch 'develop' into HAR-3157

tatut commented on src/cljc/harja/domain/oikeudet.cljc:
```diff
@@ -204,3 +204,11 @@
+  (-> kayttaja
+      :organisaatio
+      :tyyppi
+      (= tyyppi)))
```
käytä roolit/osapuoli funktiota

Jarzka replied:
Korjattu

tatut commented on src/cljs/harja/views/urakka/laadunseuranta/tarkastukset.cljs:
```diff
@@ -339,12 +340,26 @@
+            :muokattava? (constantly kenttaa-voi-muokata?)
+            :tyyppi :checkbox
+            :palstoja 2
+            :fmt fmt/totuus})
```
Piilotetaan urakoitsijalta koko checkbox, arvo palvelinpuolella aina trueksi jos käyttäjä on urakoitsija

Jarzka replied:
Korjattu

Jarzka committed:
Poista duplikaattifunktio

Jarzka committed:
Merge branch 'develop' into HAR-3157

Jarzka committed:
Merge remote-tracking branch 'origin/HAR-3157' into HAR-3157

Jarzka committed:
Piilota "näytä urakoitsijalle?" checkbox tarkastuslomakkeesta jos käyttäjä on urakoitsija

Turha näyttää valintaa joka on aina true

Jarzka committed:
Korjaa testi

Jarzka committed:
Korjaa virhe roolitarkistuksessa

Jarzka committed:
Palauta luodun tarkastuksen id oikein

Kauko committed:
Merge branch 'develop' into HAR-3157

Kauko committed:
HAR-3250: näytä urakoitsijan tarkastukset urakoitsijalle migraatiossa

Kauko committed:
Älä piirrä urakoitsijalta piilotettuja tarkastuksia karttakuvaan

Kauko committed:
Merge branch 'develop' into HAR-3157

Kauko committed:
Korjaa testien testihenkilöt, mergehärö

Kauko committed:
Merge branch 'develop' into HAR-3157

Kauko committed:
Lisää urakoitsija?-apufunktio rooleihin

Kauko committed:
Käytä uutta urakoitsija?-apufunktiota tarkastuspalveluissa

Kauko committed:
Tiestötarkastusraportti ottaa huomioon urakoitsijalta piilotetut tarkastukset

Kauko committed:
Kelitarkastusraportti ottaa huomioon urakoitsijalta piilotetut tarkastukset

Kauko committed:
Soratietarkastus ottaa huomioon urakoitsijalta piilotetut tarkastukset

Kauko committed:
Laaduntarkastusraportti ottaa huomioon urakoitsijalta piilotetut tarkastukset

Kauko committed:
Tilannekuva ottaa huomioon urakoitsijalta piilotetut tarkastukset

Kauko committed:
Merge branch 'develop' into HAR-3157

Kauko committed:
Laatupoikkeamalomake ottaa huomioon urakoitsijalta piilotetut tarkastukset

Lähinnä lisätty laatupoikkeaman-tiedot -kyselyyn JOIN tarkastustauluun,
ja sieltä tieto, näytetään lp:n tarkastusta urakoitsijalle. Lomakkeessa
sitten tarkastetaan, että jos käyttäjä on urakoitsija ja on piilotettu,
niin ei näytetä tälle siirtymänappia tarkastukseen.

Lisäksi koitin hieman parantaa ohjetekestejä asiaan liittyen.

Kauko committed:
Poista kuollutta koodia, syntynyt HAR-2879 jäljiltä

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3157

jarnovayrynen committed:
Merge branch 'develop' into HAR-3157
